### PR TITLE
Added Python code style checking

### PR DESCRIPTION
**Pull request by parsa**
This PR adds Python code styling checks to CircleCI.

### CODE REVIEW

**Comment by parsa** [comment]
@stevenrbrandt @rtohid @hkaiser I fixed some basic styling issues. Please resolve the rest.

File: tools/flake8/config.ini [hunk 1]
```diff
@@ -1,6 +1,8 @@
+[flake8]
 filename = *.py,*.py.in
 show-source = True
 statistics = True
 count = True
-ignore = F403
-max-line-length = 79
+ignore = F401,F403,F405
+max-line-length = 90
+exclude = .git,__pycache__,build,cmake
```
Comment (parsa) [inline tools/flake8/config.ini]: Why is `cmake` directory excluded? It includes `.py.in` files which need to be checked.

File: tools/flake8/config.ini [hunk 2]
```diff
@@ -1,6 +1,8 @@
+[flake8]
 filename = *.py,*.py.in
 show-source = True
 statistics = True
 count = True
-ignore = F403
-max-line-length = 79
+ignore = F401,F403,F405
```
Comment (parsa) [inline tools/flake8/config.ini]: * `F401` is for unused imports. Why disable it? People should remove unused imports.
* `F405` is mostly triggered for start references. This is to encourage people only adding the exact functions they need to the global module. Why is it disabled?

**Comment by hkaiser** [comment]
Please resolve the conflicts to enable merging.

**Comment by parsa** [comment]
Done.

**Pull request merged**